SimPEG/data_misfit.py

```diff
@@ -154,43 +199,91 @@ def W(self, value):
         self._W = value
 
     def residual(self, m, f=None):
+        r"""Computes the data residual vector for a given model.
+
+        Where :math:`\mathbf{d}_obs` is the observed data vector and :math:`\mathbf{d}_{pred}`
+        is the predicted data vector for a model vector :math:`\mathbf{m}`, this function
+        computes the data residual:
+
+        .. math::
+            \mathbf{r} = \mathbf{d}_{pred} - \mathbf{d}_{obs}
+
+        Parameters
+        ----------
+        m : (n_param, ) numpy.ndarray
+            The model for which the function is evaluated.
+        f : None, SimPEG.fields.Fields (optional)
+            A SimPEG fields object. Used when the fields for the model *m* have
+            already been computed.
+
+        Returns
+        -------
+        (n_data, ) numpy.ndarray
+            The data residual vector.
+        """
         if self.data is None:
             raise Exception("data must be set before a residual can be calculated.")
         return self.simulation.residual(m, self.data.dobs, f=f)
 
 
 class L2DataMisfit(BaseDataMisfit):
-    r"""
-    The data misfit with an l_2 norm:
+    r"""Least-squares data misfit.
+
+    Define the data misfit as the L2-norm of the weighted residual between observed
+    data and predicted data for a given model. I.e.:
 
     .. math::
+        \phi_d (\mathbf{m}) = \frac{1}{2} \big \| \mathbf{W_d}
+        \big ( \mathbf{d}_{pred} - \mathbf{d}_{obs} \big ) \big \|_2^2
+
+    where :math:`\mathbf{d}_obs` is the observed data vector, :math:`\mathbf{d}_{pred}`
+    is the predicted data vector for a model vector :math:`\mathbf{m}`, and
+    :math:`\mathbf{W_d}` is the data weighting matrix. The diagonal elements of
+    :math:`\mathbf{W_d}` are the reciprocals of the data uncertainties
+    :math:`\boldsymbol{\varepsilon}`. Thus:
 
-        \mu_\text{data} =
-            \frac{1}{2}
-            \left|
-                \mathbf{W}_d (\mathbf{d}_\text{pred} - \mathbf{d}_\text{obs})
-            \right|_2^2
+    .. math::
+        \mathbf{W_d} = diag \left ( \boldsymbol{\varepsilon}^{-1} \right )
```

Comment:
Minor change in LaTeX style:
```suggestion
        \mathbf{W_d} = \text{diag} \left ( \boldsymbol{\varepsilon}^{-1} \right )
```